portfolio/src/main/webapp/js/script.js

```diff
@@ -24,10 +24,44 @@ function responsiveTopNav() {
     }
 }
 
-function fetchJson(){
-    fetch('/data').then(response => response.text()).then((array) => {
-        console.log(array);
-        document.getElementById('fetch-result').innerText = array;
+/** Fetches the list of all user comments and builds the UI. */
+function getComments(){
+    fetch('/list-comments').then(response => response.json()).then((comments) => {
+        // Build the list of history comments.
+        const historyOfComments = document.getElementById('history');
+        console.log(comments);
+        if(comments.length == 0){
+            historyOfComments.appendChild(createListElement("OH! Be the first one to leave a message!"));
+        } else{
+            comments.forEach((entry) => {
+                console.log(entry);
```

Comment:
It usually not best practice to log user input as this could be considered personal information. Of course, its fine for this case, but just something to be aware of!

Reply:
agreed. the log was for debugging purposes. They are deleted in the new commit.